src/ApplicationCore/Services/BasketService.cs

```diff
@@ -39,15 +39,15 @@ public async Task<Basket> AddItemToBasket(string username, int catalogItemId, de
     public async Task DeleteBasketAsync(int basketId)
     {
         var basket = await _basketRepository.GetByIdAsync(basketId);
+        Guard.Against.Null(basket, nameof(basket));
         await _basketRepository.DeleteAsync(basket);
     }
 
     public async Task<Basket> SetQuantities(int basketId, Dictionary<string, int> quantities)
     {
-        Guard.Against.Null(quantities, nameof(quantities));
         var basketSpec = new BasketWithItemsSpecification(basketId);
-        var basket = await _basketRepository.GetBySpecAsync(basketSpec);
-        Guard.Against.NullBasket(basketId, basket);
+        var basket = await _basketRepository.FirstOrDefaultAsync(basketSpec);
+        Guard.Against.Null(basket, nameof(basket));
```

Comment:
In this case since the issue is that no basket is found with the given id, I think returning a Result<Basket> from the method, and a Result.NotFound in this case, would me more clear. Otherwise the error will just be that basket is null, which isn't as clear as the actual problem here.